src/units/types.cpp

```diff
@@ -1367,7 +1367,9 @@ void unit_type::apply_scenario_fix(const config& cfg)
 	}
 	if(auto attr = cfg.get("add_advancement")) {
 		for(const auto& str : utils::split(attr->str())) {
-			advances_to_.push_back(str);
+			if(std::find(begin(advances_to_), end(advances_to_), str) == std::end(advances_to_)) {
```

Comment:
i think we have an utility function `utils::contains` for this.

Reply:
@gfgtdf Ah, sure enough. Updated to use `utils::contains`